xgi/generators/nonuniform.py

```diff
@@ -288,3 +290,51 @@ def dcsbm_hypergraph(k1, k2, g1, g2, omega):
 
     df = pd.DataFrame(bipartite_edges)
     return Hypergraph(df)
+
+
+def random_hypergraph(N, ps):
+    """Generates a random hypergraph
+
+    Generate N nodes, and connect any d+1 nodes
+    by a hyperedge with probability ps[d-1].
+
+    Parameters
+    ----------
+    N : int
+        Number of nodes
+    ps : list of float
+        List of probabilities (between 0 and 1) to create a
+        hyperedge at each order d between any d+1 nodes. For example,
+        ps[0] is the wiring probability of any edge (2 nodes), ps[1]
+        of any triangles (3 nodes).
+
```

Comment:
Perhaps add a reference to the model? One relevant paper might be https://arxiv.org/abs/1703.07686